### DISCUSSION TITLE
dbus有没有运行js的接口

### DISCUSSION THREAD

**Discussion by wenfer** [linuxdeepin/developer-center#4463] [Development Questions | 开发答疑]
类似于  org.kde.kwin.Script.run 这种接口

**Comment by BLumia** [comment]
`org.kde.kwin.Script.run` 是运行 KWin Scripts 用的，意图上不是用来跑（通用的）js 的呀，我们从桌面环境的角度讲也不会提供这种功能。

你的原始需求是什么，为什么必须通过 DBus 跑 js？

**Reply by wenfer** [reply]
原始需求是获取窗口列表，和获取窗口x,y的位置，找到一个回复说要通过js才能获取到窗口列表

**Reply by BLumia** [reply]
走 kwin 的话是这样的。接口在 `org.kde.kwin.Scripting` 下。cc @justforlxz deepin-kwin 没砍掉对应的功能吧？

另外如果你只希望支持 x11 的话，不走 kwin 也可以，可以直接走 x 拿窗口属性。如果走 kwin 的接口的话肯定只能支持窗管使用 kwin 的情况。

**Reply by justforlxz** [reply]
没有删除对应功能。

**Reply by wenfer** [reply]
目前主要问题就是wayland下不好获取窗口属性，x11已经解决了
